odoo/orm/fields_relational.py

```diff
@@ -371,6 +373,56 @@ def _update_inverses(self, records, value):
                 ids1 = tuple(unique((ids0 or ()) + valid_records._ids))
                 cache.set(corecord, invf, ids1)
 
+    def condition_to_sql(self, model: BaseModel, alias: str, field_expr: str, operator: str, value, query: Query) -> SQL:
+        if operator not in ('any', 'not any') or field_expr != self.name:
+            # for other operators than 'any', just generate condition based on column type
+            return super().condition_to_sql(model, alias, field_expr, operator, value, query)
+
+        fname = field_expr
+        comodel = model.env[self.comodel_name]
+        sql_field = model._field_to_sql(alias, fname, query)
+
+        if not isinstance(value, Domain):
+            # value is SQL or Query
+            if isinstance(value, Query):
+                subselect = value.subselect()
+            elif isinstance(value, SQL):
+                subselect = SQL("(%s)", value)
+            else:
+                raise TypeError(f"condition_to_sql() 'any' operator accepts Domain, SQL or Query, got {value}")
+            sql = SQL(
+                "%s%s%s",
+                sql_field,
+                SQL(" IN ") if operator == 'any' else SQL(" NOT IN "),
+                subselect,
+            )
+            if not self.required and operator != 'any':
+                sql = SQL("(%s OR %s IS NULL)", sql, sql_field)
+            if self.company_dependent:
+                sql = self._condition_to_sql_company(sql, model, alias, field_expr, operator, value, query)
+            return sql
+
+        if self.auto_join:
+            coalias = query.make_alias(alias, self.name)
+            # auto_join bypasses checks to join the field
+            # for the comodel, the access is not bypassed, unless requested
+            if self.compute_sudo:
```

Comment:
I don't get it. Why the compute_sudo flag of the current many2one allow you to make a sudo on the comodel ?
Do you have a example of it ?

Reply:
It seems we do not have that case. It would require a computed (with sudo) auto_join field, which I think makes little sense. In the compute function sudo would be set, so I think that it is ok to have the sudo here.